pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -131,6 +136,7 @@ func Create(cli client.Client, installation *operatorv1.InstallationSpec, cluste
 	var (
 		cryptoCA                      *crypto.CA
 		csrImage                      string
+		privateKey                    any
```

Comment:
The use of `any` comes from how it is defined in golang's x509 package. It predates the generics, my guess is that otherwise the implementation would have been different.
The x509 functions and methods need `any` to be passed in. At the time of writing, the three supported private key types are rsa, ecdsa, ed25519. We could implement a new interface containing these constraints, but it would introduce more code + casting / parsing, without any practical implications that I can think of. If someone goes out of his/her way to replace the tigera-ca-private secret with a broken private key, we can still leverage the same code in the x509 package to return the right error.

I'll add a comment.